### PR TITLE
chore: schematics - extract the lib generating mechanism

### LINKED ISSUE #9996: Schematics - improve the mechanism for adding feature libs
Right now we have a lot of common code in both `misc` and `organization` libraries, that can be extracted to the `projects/schematics` and re-used in both libraries, as well as in the future libraries.

### PR DESCRIPTION
**Pull request by znikola**
closes #9996

### CODE REVIEW

File: projects/schematics/src/add-pwa/index.ts [hunk 1]
```diff
@@ -6,12 +6,11 @@ import {
   SchematicsException,
   Tree,
 } from '@angular-devkit/schematics';
+import { of } from '@angular-devkit/schematics/node_modules/rxjs';
```
Comment (marlass) [inline projects/schematics/src/add-pwa/index.ts:9]: 👀

File: feature-libs/misc/schematics/add-misc/index.ts [hunk 2]
```diff
@@ -117,340 +69,26 @@ function addStorefinderFeature(
...
-      },
-    })`
-    );
-
-    changes.push(resourceImportChange, chunkImportChange, ...providersChanges);
-  }
-
-  return changes;
+    },
+    {
```
Comment (marlass) [inline feature-libs/misc/schematics/add-misc/index.ts]: Do we have a dependency from @spartacus/misc to @spartacus/setup? It looks for me like some kind of leftover.

File: package.json [hunk 3]
```diff
@@ -117,6 +117,7 @@
     "@angular/platform-browser": "^10.1.0",
     "@angular/platform-browser-dynamic": "^10.1.0",
     "@angular/platform-server": "^10.1.0",
+    "@angular/pwa": "^0.1001.0",
```
Comment (marlass) [inline package.json:120]: It seems unrelated to the changes

File: projects/schematics/src/shared/utils/lib-utils_spec.ts [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+      const rule = addLibraryFeature(
+        appModulePath,
+        BASE_OPTIONS,
+        BASE_FEATURE_CONFIG
+      );
+      const result = await schematicRunner.callRule(rule, appTree).toPromise();
+
+      const appModule = result.read(appModulePath).toString(UTF_8);
+      expect(appModule).toContain(`resources: ${I18N_RESOURCES},`);
+      expect(appModule).toContain(`chunks: ${I18N_CHUNKS},`);
```
Comment (marlass) [inline projects/schematics/src/shared/utils/lib-utils_spec.ts:144]: We should also check imports here

File: projects/schematics/src/shared/utils/html-utils.ts [hunk 5]
```diff
@@ -0,0 +1,123 @@
+import { SchematicsException, Tree } from '@angular-devkit/schematics';
```
Comment (marlass) [inline projects/schematics/src/shared/utils/html-utils.ts:1]: Can you also copy spec file for this utils?

File: projects/schematics/package.json [hunk 6]
```diff
@@ -18,14 +18,13 @@
...
-    "@angular-devkit/core": "^10.1.0",
-    "@angular-devkit/schematics": "^10.1.0",
-    "@angular/cdk": "^10.2.0",
     "@angular/localize": "^10.1.0",
+    "@nguniversal/express-engine": "^10.1.0"
+  },
+  "peerDependencies": {
     "@angular/pwa": "^0.1001.0",
-    "@nguniversal/express-engine": "^10.1.0",
-    "@schematics/angular": "^10.1.0",
```
Comment (marlass) [inline projects/schematics/package.json:27]: Why we removed the @schematics/angular ? I still see a lot of references to it.

**Review by marlass — CHANGES_REQUESTED** [review]

File: projects/schematics/package.json (hunk 6 above)
Comment (znikola) [reply projects/schematics/package.json:27]: We're pulling it from the parent node_modules. I added it to the peerDeps now.

File: projects/schematics/src/shared/utils/html-utils.ts (hunk 5 above)
Comment (znikola) [reply projects/schematics/src/shared/utils/html-utils.ts:1]: There's no spec file.

Comment (znikola) [reply projects/schematics/src/shared/utils/html-utils.ts:1]: https://github.com/angular/components/blob/master/src/cdk/schematics/utils/html-manipulation.ts

File: package.json (hunk 3 above)
Comment (znikola) [reply package.json:120]: We discussed it offline - it's related to https://github.com/SAP/spartacus/pull/10003#discussion_r531575698

File: projects/schematics/src/shared/utils/lib-utils_spec.ts (hunk 4 above)
Comment (znikola) [reply projects/schematics/src/shared/utils/lib-utils_spec.ts:144]: Done

File: feature-libs/misc/schematics/add-misc/index.ts (hunk 2 above)
Comment (znikola) [reply feature-libs/misc/schematics/add-misc/index.ts]: Nice catch, it's a left over.

**Review by marlass — APPROVED** [review]
